closed/adds/jdk/src/share/classes/jdk/crypto/jniprovider/NativeCrypto.java

```diff
@@ -54,32 +54,44 @@ public class NativeCrypto {
     private static final boolean traceEnabled = Boolean.parseBoolean(
             GetPropertyAction.privilegedGetProperty("jdk.nativeCryptoTrace", "false"));
 
-    //ossl_vers will be either:
+    //ossl_ver will be either:
     // -1 : library load failed
     // or one of the OPENSSL_VERSION_x_x_x constants
-    private static final boolean loaded = AccessController.doPrivileged(
-            (PrivilegedAction<Boolean>) () -> {
-            Boolean isLoaded = Boolean.FALSE;
-
+    private static final long ossl_ver = AccessController.doPrivileged(
+            (PrivilegedAction<Long>) () -> {
+            long osslVersion;
             try {
                 System.loadLibrary("jncrypto"); // check for native library
                 // load OpenSSL crypto library dynamically.
-                long ossl_ver = loadCrypto(traceEnabled);
-                if (ossl_ver != -1) {
-                    isLoaded = Boolean.TRUE;
+                osslVersion = loadCrypto(traceEnabled);
+                if (traceEnabled && (osslVersion != -1)) {
+                    System.err.println("Native crypto library load succeeded - using native crypto library.");
                 }
             } catch (UnsatisfiedLinkError usle) { 
                 if (traceEnabled) {
                         System.err.println("UnsatisfiedLinkError: Failure attempting to load jncrypto JNI library");
+                        System.err.println("Warning: Native crypto library load failed." +
+                                " Using Java crypto implementation.");
```

Comment:
Please correct the indentation of these lines (4 spaces too many on each line).

Also, while you're here, please remove the trailing whitespace on lines 70, 79 and 82.

Reply:
Done.